src/farcasterd/runtime.rs

```diff
@@ -513,39 +527,53 @@ impl Runtime {
-                    );
-                    let resp = self.listen(&remote_addr);
-                    match resp {
-                        Ok(_) => info!(
-                            "Connection daemon {} for incoming LN peer connections on {}",
-                            "listens".bright_green_bold(),
-                            remote_addr
-                        ),
-                        Err(ref err) => error!("{}", err.err()),
+            Request::MakeOffer(request::ProtoPublicOffer {
+                offer,
+                remote_addr,
+                peer_secret_key,
+            }) => {
+                let resp = match (self.listens.contains(&remote_addr), peer_secret_key) {
+                    (false, None) => {
+                        trace!("Push MakeOffer to pending_requests and requesting a secret from Wallet");
+                        return self.get_secret(senders, source, request);
                     }
-
-                    report_to.push((
+                    (false, Some(sk)) => {
+                        self.listens.insert(remote_addr);
+                        info!(
+                            "{} for incoming LN peer connections on {}",
+                            "Starting listener".bright_blue_bold(),
+                            remote_addr.bright_blue_bold()
+                        );
+                        self.listen(&remote_addr, sk)
+                    }
+                    (true, _) => {
+                        info!("Already listening on {}", &remote_addr);
+                        let msg = format!("Already listening on {}", &remote_addr);
+                        Ok(msg)
+                    }
+                };
+                match resp {
+                    Ok(_) => info!(
+                        "Connection daemon {} for incoming LN peer connections on {}",
+                        "listens".bright_green_bold(),
+                        remote_addr
+                    ),
+                    Err(err) => {
+                        error!("{}", err.err());
+                        return Err(err);
+                    }
+                }
+                report_to.push((
                         Some(source.clone()),
                         resp.into_progress_or_failure()
                         // Request::Progress(format!(
                         //     "Node {} listens for connections on {}",
                         //     self.node_id, remote_addr
                         // )),
                     ));
-                } else {
-                    info!("Already listening on {}", &remote_addr);
-                    let msg = format!("Already listening on {}", &remote_addr);
-                    report_to.push((Some(source.clone()), Request::Progress(msg)));
-                }
+
                 let peer = internet2::RemoteNodeAddr {
-                    node_id: self.node_id()?,
+                    node_id: self.node_ids()[0], // FIXME shouldnt be 0 here
```

Comment:
Does that FIXME should be handle now?

Reply:
#922ce4d reports an error if there are multiple node_ids stating multiple ids are not supported